integration/test_db.py

```diff
@@ -0,0 +1,244 @@
+        assert testref.key == 'dinodb'
+        assert testref.path == '/_adminsdk/python/dinodb'
+
+    def test_child(self, testref):
+        child = testref.child('dinosaurs')
+        assert child.key == 'dinosaurs'
+        assert child.path == '/_adminsdk/python/dinodb/dinosaurs'
+
+    def test_parent(self, testref):
+        parent = testref.parent
+        assert parent.key == 'python'
+        assert parent.path == '/_adminsdk/python'
+
+
+class TestReadOperations(object):
+    """Test cases for reading node values."""
+
+    def test_get_value(self, testref, testdata):
+        value = testref.get_value()
+        assert isinstance(value, dict)
+        assert testdata == value
+
+    def test_get_child_value(self, testref, testdata):
+        value = testref.child('dinosaurs').get_value()
+        assert isinstance(value, dict)
+        assert testdata['dinosaurs'] == value
+
+    def test_get_grandchild_value(self, testref, testdata):
+        value = testref.child('dinosaurs').child('lambeosaurus').get_value()
+        assert isinstance(value, dict)
+        assert testdata['dinosaurs']['lambeosaurus'] == value
+
+    def test_get_nonexisting_child_value(self, testref):
+        assert testref.child('none_existing').get_value() is None
+
+
+class TestWriteOperations(object):
+    """Test cases for creating and updating node values."""
+
+    def test_push(self, testref):
+        python = testref.parent
+        ref = python.child('users').push()
+        assert ref.path == '/_adminsdk/python/users/' + ref.key
+        assert ref.get_value() == ''
+
+    def test_push_with_value(self, testref):
+        python = testref.parent
+        value = {'name' : 'Luis Alvarez', 'since' : 1911}
+        ref = python.child('users').push(value)
+        assert ref.path == '/_adminsdk/python/users/' + ref.key
+        assert ref.get_value() == value
+
+    def test_set_primitive_value(self, testref):
+        python = testref.parent
+        ref = python.child('users').push()
+        ref.set_value('value')
+        assert ref.get_value() == 'value'
+
+    def test_set_complex_value(self, testref):
+        python = testref.parent
+        value = {'name' : 'Mary Anning', 'since' : 1799}
+        ref = python.child('users').push()
+        ref.set_value(value)
+        assert ref.get_value() == value
+
+    def test_set_primitive_value_with_priority(self, testref):
+        python = testref.parent
+        ref = python.child('users').push()
+        ref.set_value('value', 1)
+        assert ref.get_value() == 'value'
+        assert ref.get_priority() == 1
+
+    def test_set_complex_value_with_priority(self, testref):
+        python = testref.parent
+        value = {'name' : 'Barnum Brown', 'since' : 1873}
+        ref = python.child('users').push()
+        ref.set_value(value, 2)
+        assert ref.get_value() == value
+        assert ref.get_priority() == 2
+
+    def test_update_children(self, testref):
```

Comment:
You might consider a test for update_children that updates an existing value, preserving attributes not in the value you pass.